core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java

```diff
@@ -184,4 +190,23 @@ protected String getEnvelopName() {
     protected String getBaseTypeName() {
         throw new UnsupportedOperationException();
     }
+    
+    private String normalizeInstrumentationKey(String instrumentationKey){
+    	if (StringUtils.isNullOrEmpty(instrumentationKey)){
+    		return "";
+    	}
+    	else{
+    		return instrumentationKey.replace("-", "").toLowerCase() + ".";
```

Comment:
@XiaoliangHuang if in rare case instrumentation key is only "----" (dashes) then this part will introduce additional dot. I don't think this is desirable. Please add a check and a test case also.

Reply:
Add special case handler, if there are only characters from "-/.", then do not contains ikey.
maybe in the future, should add some schema check on the configuration to restrict the usage of special characters in the ikey